core/integration/module_shell_test.go

```diff
@@ -620,6 +624,15 @@ func (ShellSuite) TestArgsSpread(ctx context.Context, t *testctx.T) {
 	}
 }
 
+func (ShellSuite) TestCommandStateArgs(ctx context.Context, t *testctx.T) {
+	c := connect(ctx, t)
+	script := fmt.Sprintf("FOO=$(container | from %s | with-exec -- echo -n foo | stdout); .doc $FOO", alpineImage)
```

Comment:
Does something like `FOO=$(container | from %s); $FOO | with-exec -- echo -n foo | stdout` also work now?

Reply:
Here it is!
- https://github.com/dagger/dagger/pull/9191